src/module/item/spellcasting-entry/collection.ts

```diff
@@ -107,11 +113,11 @@ class SpellCollection<TActor extends ActorPF2e, TEntry extends BaseSpellcastingE
     }
 
     /** Saves the prepared spell slot data to the spellcasting entry  */
-    async prepareSpell(spell: SpellPF2e, slotRank: number, spellSlot: number): Promise<TEntry | undefined> {
+    async prepareSpell(spell: SpellPF2e, groupId: SpellSlotGroupId, slotId: number): Promise<TEntry | undefined> {
         this.#assertEntryIsDocument(this.entry);
 
-        if (spell.baseRank > slotRank && !(slotRank === 0 && spell.isCantrip)) {
-            const targetRankLabel = game.i18n.format("PF2E.Item.Spell.Rank.Ordinal", { rank: ordinalString(slotRank) });
+        if (groupId !== "cantrips" && spell.baseRank > groupId) {
```

Comment:
So I think before you couldn't prepare cantrips in non-cantrip slots. Is this still true after this change?